network/src/peers/peer/connector.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    io::{Error as IoError, ErrorKind},
+    time::Duration,
+};
+
+use futures::{select, FutureExt};
+use snarkvm_objects::Storage;
+use tokio::{net::TcpStream, sync::mpsc};
+
+use super::PeerAction;
+use crate::{stats, NetworkError, Node, Peer, PeerEvent, PeerEventData, PeerHandle, Version};
+
+use super::network::PeerNetwork;
+
+const CONNECTION_TIMEOUT_SECS: u64 = 3;
+
+impl Peer {
+    async fn inner_connect(&mut self, our_version: Version) -> Result<PeerNetwork, NetworkError> {
```

Comment:
this name could be more informative, e.g. `perform_handshake`

Reply:
the intent is to abstract the tcp connection and handshake. this function also does more than just initiate the handshake, it starts a TCP connection it completes the handshake. it literally `connect`s.